packages/cfg-core/lib/nodes/Node.ts

```diff
@@ -22,14 +22,14 @@
  * @author Dimitri Stallenberg
  */
 export interface Node {
-  type: NodeType;
+  readonly type: NodeType;
 
-  id: string;
+  readonly id: string;
 
-  lines: number[];
-  statements: string[];
+  readonly lines: readonly number[];
```

Comment:
Why is readonly used twice here?

Reply:
It doesn't prohibit you from writing explicitly in code `nodes.at(0).lines = [23]` as that field is not readonly then, and so redefinition of the entire list is possible.